internal/addrs/parse_ref.go

```diff
@@ -478,3 +494,13 @@ func parseSingleAttrRef(traversal hcl.Traversal) (string, hcl.Range, hcl.Travers
 	})
 	return "", hcl.Range{}, nil, diags
 }
+
+func appendRunDiag(subject *hcl.Range) tfdiags.Diagnostics {
+	var diags tfdiags.Diagnostics
+	return diags.Append(&hcl.Diagnostic{
+		Severity: hcl.DiagError,
+		Summary:  "Invalid reference",
+		Detail:   `The "run" block output values must be followed by two attribute names: the resource type and the property. For example: run.resource_type.property.`,
```

Comment:
I don't believe this error is correct. We are not referencing a resource + property of the `run` block. We are referencing and output in the format of `run.aws_access_key_id`, which means that after the `run` we have the run name and the name of the output.